### ISSUE TITLE
5.x version failed to compile using AOT

### ISSUE THREAD

**Issue by gogoout** [triniwiz/nativescript-stripe#33]
### Please, tell us how to recreate the issue in as much detail as possible. 
Just build it with --env.aot . After I switch back to 4.X. The problem disappeared.
The error message is something  like "Please add a @NgModule annotation" .

**Comment by triniwiz** [comment]
Thanks for the info

**Comment by RobertGardner** [comment]
I am able to successfully build my app with --env.aot and the 5.x version. My app is an Angular app. Is yours?
Do you have a sample app I can access that will recreate the problem?

**Comment by gogoout** [comment]
Emm, I tried switching back to 5.x again, and had the same error

Here is my import code
```js
import {CreditCardViewModule} from 'nativescript-stripe/angular';
@NgModule({
    imports: [...SHARED_MODULES, DropDownModule, TNSCheckBoxModule, CreditCardViewModule],
   ...
})

```

```
ERROR in : Unexpected value 'CreditCardViewModule in /path/AllGoodsX/node_modules/nativescript-stripe/angular/index.js' imported by the module 'CoreModule in /path/AllGoodsX/public/src/app/core/core.module.ts'. Please add a @NgModule annotation.
```

I noticed that in `nativescript-stripe/angular` folder there are no *.d.ts files while normally other plugin have.

@RobertGardner  Have you tried --env.uglify --env.aot? That's how I build mine. Also my typescript version is `2.9.2`. Don't know if that helps.

Unfortunately, I can't provide a simple project now, my one is quite a large project which is already released into Playstore and Appstore. 
I can try to build a small project when I have time.

**Comment by RobertGardner** [comment]
Both of the demo apps (`demo` and `demo-angular`) build with `--env.aot --env.uglify`.

Can you build and run without `--env.aot`?

It looks like you are importing the module correctly.

This is a long shot, but I wonder if you are picking up the 4.x version of `nativescript-stripe.module.js` or some other file. The 5.x version has this code at line 9:

```
    CreditCardViewModule = __decorate([
        core_1.NgModule({
            declarations: [nativescript_stripe_directives_1.DIRECTIVES],
            exports: [nativescript_stripe_directives_1.DIRECTIVES],
        })
    ], CreditCardViewModule);
```
so you can see that it is annotated with `@NgModule`. Have you tried deleting your `node_modules` folder between builds? You could also look in your `node_modules` folder to see if that code is there.

Sorry, that's all I can think of trying without a sample that reproduces the problem.

(There is an index.d.ts file at the root. If you follow the [instructions](https://docs.nativescript.org/plugins/angular-plugin) for supporting Angular in a plugin, you don't end up with a .d.ts file in the /angular folder.)

**Comment by triniwiz** [comment]
It can be the way the lib was published i'll take a look

**Comment by gogoout** [comment]
Yes, I can confirm the code is there.  And I can run project correctly without `--env.aot`. Just to clarify all the files in angular folder is
`index.js   nativescript-stripe.directives.js   nativescript-stripe.module.js     package.json`

**Comment by RobertGardner** [comment]
That all looks correct. It is probably not a packaging issue. 

It will be difficult to debug without a sample I can work with, but I'll try a few things and see if anything jumps out. This will take me a few days since my "real job" is calling.

**Comment by gogoout** [comment]
I can see this bug maybe more complicated than I expected. Maybe it's having something to do with my project only. I have tried server ways that I think might lead to the error but haven't got luck yet.  
But it's not a serious error either. If I registerElement the old way, I can still compile it with Aot. So If you guys can't reproduce it, you can just close the issues I guess.

**Comment by RobertGardner** [comment]
I have tried to recreate this on several apps and have not been successful. These apps are all running NativeScript 4 or 5, so maybe the problem is with earlier versions?

Since I can't reproduce it, I'm going to close this issue. Feel free to open it again if you have more info or ideas on how to reproduce.

**Comment by vishalpandeyy** [comment]
> registerElement



> I can see this bug maybe more complicated than I expected. Maybe it's having something to do with my project only. I have tried server ways that I think might lead to the error but haven't got luck yet.
> But it's not a serious error either. If I registerElement the old way, I can still compile it with Aot. So If you guys can't reproduce it, you can just close the issues I guess.

I have this issue as well.. can you please tell me how to use registerElement the old way to be able to compile it with aot? 

Thank you

**Comment by gogoout** [comment]
@vishalpandeyy 
```typescript
registerElement('CreditCardView', () => require('nativescript-stripe').CreditCardView);
```

**Comment by jeremypele** [comment]
I can confirm that this issue is not an isolated one, have the same when running with `--env.aot`.

**Comment by RobertGardner** [comment]
@triniwiz I'm also having this trouble, compiling with `ns-stripe v5.4.2` on my personal Angular app. I'm using `tns v5.4.2` running with the new bundle workflow. When I run
`tns run ios --emulator --env.aot`

I get the error: 
```
ERROR in : Unexpected value 'CreditCardViewModule in <...>node_modules/nativescript-stripe/angular/nativescript-stripe.module.d.ts' imported by the module 'AppModule in <...>src/app/app.module.ts'. Please add a @NgModule annotation.
```

It runs fine without `--env.aot`.

Did I mess up with my deployment? I used the process in `/publish/instructions.md`.

Will this all change with NS v6?

Sorry I don't know enough to debug this myself.

**Comment by jeremypele** [comment]
@RobertGardner @triniwiz The issue is coming from the missing `xxx.metadata.json` missing in the npm version of the plugin. Angular AOT requires it. Generated one from the source code, pasted it into my node_modules and it's working just fine.

**Comment by RobertGardner** [comment]
Thanks! This was my mistake. @triniwiz provided the correct build scripts, but I failed to update my publish scripts accordingly. I've published a new version (5.4.3) that contains the metadata files. Hopefully this is now fixed permanently! ;-)
